Rakefile

```diff
@@ -39,6 +39,38 @@ task :rubygems_admins do
   end
 end
 
+task :update_dependencies do
+  puts "Updating all internal fastlane dependencies"
+
+  regex = %r{spec.add_dependency .(.+).\, .\>\= (\d+\.\d+\.\d+).\, .\< \d+\.\d+\.\d+.}
+
+  Dir["./**/*.gemspec"].each do |current_gemspec_path|
+    content = File.read(current_gemspec_path)
+
+    content.gsub!(regex) do |full_match|
+      matched = full_match.match(regex)
+      gem_name = matched[1]
+      current_version_number = Gem::Version.new(matched[2]) # used to detect if we actually changed something
```

Comment:
According to the docs for `gsub` you can use `$1` and `$2` to refer to these match values rather than running the regex match again.